prbt_hardware_support/test/integrationtests/integrationtest_execute_brake_test.cpp

```diff
@@ -0,0 +1,147 @@
+ */
+
+#include <gtest/gtest.h>
+
+#include <ros/ros.h>
+
+#include <prbt_hardware_support/BrakeTest.h>
+#include <prbt_hardware_support/WriteModbusRegister.h>
+#include <prbt_hardware_support/canopen_chain_node_mock.h>
+#include <prbt_hardware_support/joint_states_publisher_mock.h>
+#include <prbt_hardware_support/modbus_api_spec.h>
+#include <prbt_hardware_support/pilz_manipulator_mock.h>
+#include <prbt_hardware_support/pilz_modbus_server_mock.h>
+#include <prbt_hardware_support/ros_test_helper.h>
+
+static const std::string EXECUTE_BRAKE_TEST_SERVICE_NAME{"/prbt/execute_braketest"};
+static const std::string MODBUS_WRITE_SERVICE_NAME{"/pilz_modbus_client_node/modbus_write"};
+static const std::string CONTROLLER_HOLD_MODE_SERVICE_NAME{"/prbt/manipulator_joint_trajectory_controller/hold"};
+static const std::string CONTROLLER_UNHOLD_MODE_SERVICE_NAME{"/prbt/manipulator_joint_trajectory_controller/unhold"};
+
+static constexpr double WAIT_FOR_BRAKE_TEST_SERVICE_TIMEOUT_S{5.0};
+static constexpr uint16_t MODBUS_BRAKE_TEST_PREPARE_VALUE {0};
+static constexpr uint16_t MODBUS_BRAKE_TEST_EXPECTED_VALUE {1};
+
+/**
+ * @tests{Execute_BrakeTest_mechanism,
+ *  Test the BrakeTest service node.
+ * }
+ *
+ * Test Sequence:
+ *  0. Preparing modbus mock
+ *  1. Initialize CANOpen mock, JointStatesPublisher mock and manipulator mock
+ *  2. Wait for BrakeTest service
+ *  3. Make the service call
+ *  4. Check contents of relevant modbus registers
+ *  5. Shutdown of modbus mock
+ *
+ * Expected Results:
+ *  0. -
+ *  1. -
+ *  2. Service is available
+ *  3. Service call is successful
+ *  4. Brake test success was set in modbus
+ *  5. -
+ */
+TEST(IntegrationtestExecuteBrakeTest, testBrakeTestService)
+{
+  using namespace prbt_hardware_support;
+  ros::NodeHandle nh;
+  ros::NodeHandle nh_private("~");
+
+  /**********
+   * Step 0 *
+   **********/
+  std::string ip;
+  int port;
+  ASSERT_TRUE(nh_private.getParam("modbus_server_ip", ip));
+  ASSERT_TRUE(nh_private.getParam("modbus_server_port", port));
+
+  ModbusApiSpec write_api_spec {nh, "write_api_spec/"};
+  unsigned int modbus_register_size {write_api_spec.getMaxRegisterDefinition() + 1U};
+  prbt_hardware_support::PilzModbusServerMock modbus_server(modbus_register_size);
+  std::thread modbus_server_thread( &initalizeAndRun<prbt_hardware_support::PilzModbusServerMock>,
+                                    std::ref(modbus_server), ip.c_str(), static_cast<unsigned int>(port) );
+  ASSERT_TRUE(write_api_spec.hasRegisterDefinition(modbus_api_spec::BRAKETEST_PERFORMED));
+  unsigned int register_perf = write_api_spec.getRegisterDefinition(modbus_api_spec::BRAKETEST_PERFORMED);
+  ASSERT_TRUE(write_api_spec.hasRegisterDefinition(modbus_api_spec::BRAKETEST_RESULT));
+  unsigned int register_res = write_api_spec.getRegisterDefinition(modbus_api_spec::BRAKETEST_RESULT);
+
+  modbus_server.setHoldingRegister({{register_perf, MODBUS_BRAKE_TEST_PREPARE_VALUE}});
```

Comment:
Should we provide a function that blocks until the value is set so that wo don't need the retries in the upcoming lines?

Reply:
The Idea was to not depend on a very long timeout but still be able to finish faster, if possible